sdk/extensions/Azure.Extensions.AspNetCore.DataProtection.Keys/src/AzureDataProtectionKeyVaultKeyBuilderExtensions.cs

```diff
@@ -20,6 +20,19 @@ namespace Microsoft.AspNetCore.DataProtection
     /// </summary>
     public static class AzureDataProtectionKeyVaultKeyBuilderExtensions
     {
+        /// <summary>
+        /// Configures the data protection system to protect keys with specified key in Azure KeyVault.
+        /// </summary>
+        /// <param name="builder">The builder instance to modify.</param>
+        /// <param name="keyIdentifier">The Azure Key Vault key identifier used for key encryption.</param>
+        /// <param name="tokenCredential">The token credential to use for authentication.</param>
+        /// <returns>The value <paramref name="builder"/>.</returns>
+        public static IDataProtectionBuilder ProtectKeysWithAzureKeyVault(this IDataProtectionBuilder builder, string keyIdentifier, TokenCredential tokenCredential)
+        {
+            Argument.AssertNotNull(tokenCredential, nameof(tokenCredential));
```

Comment:
I'm assuming this was left out originally because `KeyResolver` also performs a null check on the parameter.   I like adding it here for better locality in the stack trace, if nothing else.

Reply:
Added all null-checks back